### PR TITLE
updated the core script to get distinct connectors

### PR DESCRIPTION
**Pull request by Dragos0000**

### CODE REVIEW

File: src/owl-core-lib/connectors-owl-core.xsl [hunk 1]
```diff
@@ -24,6 +24,7 @@
     <xsl:import href="../common/fetchers.xsl"/>
     <xsl:import href="../common/utils.xsl"/>
     <xsl:import href="../common/formatters.xsl"/>
+    <xsl:import href="../html-conventions-lib/utils-html-conventions.xsl"/>
```
Comment (costezki) [inline src/owl-core-lib/connectors-owl-core.xsl]: why do we need html-conventions-utils? 

If we do then those utils shall be somewhere else

File: src/owl-core-lib/connectors-owl-core.xsl [hunk 2]
```diff
@@ -93,107 +94,90 @@
...
+                for $connector in $connectorsWithSameName
+                return
+                    if ($connector/documentation/@value) then
+                        fn:concat($connector/documentation/@value, ' (', f:getConnectorName($connector), ') ')
+                    else
+                        ()"/>
 
-        <xsl:variable name="connectorDocumentation"
-            select="f:formatDocString(./documentation/@value)"/>
+        <xsl:variable name="documentation"
```
Comment (costezki) [inline src/owl-core-lib/connectors-owl-core.xsl:152]: can we add connector name here as well, to trace the sources as annotations, just like in the case of attribute definition aggregation.

File: src/owl-core-lib/connectors-owl-core.xsl [hunk 3]
```diff
@@ -93,107 +94,90 @@
...
-                        <xsl:value-of select="$note"/>
-                    </skos:scopeNote>
-                    <rdfs:comment xml:lang="en">
-                        <xsl:value-of select="$note"/>
-                    </rdfs:comment>
-                </xsl:if>
-                <rdfs:isDefinedBy rdf:resource="{$base-uri}"/>
-            </owl:ObjectProperty>
-        </xsl:if>
+        <xsl:variable name="note" select="f:formatDocString(fn:string-join($connectorNotes))"/>
```
Comment (costezki) [inline src/owl-core-lib/connectors-owl-core.xsl:167]: by the way!!!

All the definitions/notes from any element (connector, attribute, class, enumeration) must be filtered/formatted using the function `f:formatDocString`. The reason is funky formatting used by the EA and it is not according to HTML standards and could lead to encoding problems.

File: src/owl-core-lib/connectors-owl-core.xsl [hunk 4]
```diff
@@ -93,107 +94,90 @@
...
-                        <xsl:value-of select="$note"/>
-                    </skos:scopeNote>
-                    <rdfs:comment xml:lang="en">
-                        <xsl:value-of select="$note"/>
-                    </rdfs:comment>
-                </xsl:if>
-                <rdfs:isDefinedBy rdf:resource="{$base-uri}"/>
-            </owl:ObjectProperty>
-        </xsl:if>
+        <owl:ObjectProperty rdf:about="{$roleURI}">
```
Comment (costezki) [inline src/owl-core-lib/connectors-owl-core.xsl:171]: very nice simplification

**Review by costezki — APPROVED** [review]
need to check teh usage od `formattDocumentation` function

**Review by costezki — CHANGES_REQUESTED** [review]
I just checked the RDF loading in TopBraid and it fails for the following reason:

`[line: 9170, col: 19] {E202} Expecting XML start or end element(s). String data "Default for " not allowed ...`

This means that somewhere the Default rule kicks in and it should not. A quick fix would be to change the selectors OR create a `match` template with empty output

File: src/owl-core-lib/connectors-owl-core.xsl (hunk 1 above)
Comment (Dragos0000) [reply src/owl-core-lib/connectors-owl-core.xsl]: moved functions to fetchers

File: src/owl-core-lib/connectors-owl-core.xsl (hunk 3 above)
Comment (Dragos0000) [reply src/owl-core-lib/connectors-owl-core.xsl:167]: created new ticket

**Comment by Dragos0000** [comment]
retest the RDF loading in TopBraid

**Review by costezki — APPROVED** [review]
can but need further investigations.

**Review by costezki — APPROVED** [review]
well done, thank you

**Pull request merged**